cmd/vulcan-retirejs/main.go

```diff
@@ -99,77 +108,103 @@ func runRetireJs(ctx context.Context, args []string) ([]RetireJsFileResult, erro
 		args = retireArgs
 	}
 	var report RetireJsReport
-	noFindings, findings, _, err := command.ExecuteWithStdErr(ctx, logger, args[0], args[1:]...)
-	if err != nil {
-		return report.Data, err
-	}
+	_, err := command.ExecuteAndParseJSON(ctx, logger, &report, args[0], args[1:]...)
 
-	if len(findings) > 0 {
-		err = json.Unmarshal(findings, &report)
-	} else {
-		err = json.Unmarshal(noFindings, &report)
-	}
 	return report.Data, err
 }
 
 func addVulnsToState(state checkstate.State, r []RetireJsFileResult) {
-	vulns := make(map[string]report.Vulnerability)
-	for _, finding := range r {
-		for _, result := range finding.Results {
-			for _, vuln := range result.Vulnerabilities {
-				summaryText := vuln.Identifiers.Summary
-				if vuln.Identifiers.Summary == "" {
-					summaryText = "Vulnerabilities in JavaScript Dependencies"
-				}
-
-				v, ok := vulns[summaryText]
-				if !ok {
-					v = report.Vulnerability{
-						Summary: summaryText,
-						Description: "Vulnerabilities in dependencies may impact in the security of your program. For that reason " +
-							"it's important to check for issues not only in your code but in the 3rd party code you are using as a dependency.",
-						Score: 0.0,
-						Recommendations: []string{
-							fmt.Sprintf("Check if there is an update available for the affected components."),
-							"Additional vulnerability information can be found in the links in resources",
-						},
-						Resources: []report.ResourcesGroup{
-							report.ResourcesGroup{
-								Name: "Vulnerable Components",
-								Header: []string{
-									"Component",
-									"Version",
-								},
-							},
+	for _, f := range r {
+		for _, v := range f.Results {
+			fingerprint := []string{}
+			vulnerability := report.Vulnerability{
+				Summary: "Vulnerabilities in JavaScript Dependencies",
+				CWEID:   1104,
+				Labels:  []string{"potential", "web", "retirejs"},
+				Description: "Vulnerabilities in dependencies may impact in the security of your program. For that reason " +
+					"it's important to check for issues not only in your code but in the 3rd party code you are using as a dependency.",
+				Recommendations: []string{
+					fmt.Sprintf("Check if there is an update available for the affected resource."),
+					"Additional vulnerability information can be found in the links in the resources table.",
+				},
+				References:       []string{"https://portswigger.net/kb/issues/00500080_vulnerable-javascript-dependency"},
+				AffectedResource: fmt.Sprintf("%s-%s", v.Component, v.Version), // example: jquery-1.9.0
+				Score:            0.0,
+				Resources: []report.ResourcesGroup{
+					{
+						Name: "Vulnerabilities",
+						Header: []string{
+							"Dependency",
```

Comment:
Do you think we could remove the `Dependency` column?

Reply:
Agree, this column contains/shows redundant information.
The column has been removed here [26dc483](https://github.com/adevinta/vulcan-checks/pull/145/commits/26dc4835c451fccac662ae6b416db187e45083fb)